### PR TITLE
Example Dataset - Spatial Datasets

### LINKED ISSUE #814: Update spatial analysis notebooks to use Huggingface dataset
**Is your feature request related to a problem? Please describe.**
The current spatial analysis notebooks use random example data in the repo. These datasets are generally too small to be useful for understanding the output of these analyses, and don't have the same format as our pipeline

**Describe the solution you'd like**
Update these notebooks to use the huggingface datasets instead. None of these notebooks should require a new dataset from huggingface, as they just require a cell table, the image data, and segmentation labels. The main thing that will need to be done is make modifications to the expected directory structure and input data to align with our current scheme.

### PR DESCRIPTION
**Pull request by srivarra**
**If you haven't already, please read through our [contributing guidelines](https://ark-analysis.readthedocs.io/en/latest/_rtd/contributing.html) before opening your PR**

**What is the purpose of this PR?**

Closes #814.

**How did you implement your changes**

Added datasets for:
- Example Fiber Segmentation
- Example Neighborhood Analysis
- Example Pairwise Spatial Enrichment
- LDA Preprocessing
- LDA Training and Inference

**Remaining issues**

- [x] Merge the hugging face branch to main.

### CODE REVIEW

File: ark/utils/example_dataset_test.py [hunk 1]
```diff
@@ -1,12 +1,21 @@
...
 
 
-@pytest.fixture(scope="session", params=["segment_image_data", "cluster_pixels",
-                                         "cluster_cells", "post_clustering"])
+@pytest.fixture(scope="session", params=["segment_image_data",
+                                         "cluster_pixels",
+                                         "cluster_cells",
+                                         "post_clustering",
+                                         "fiber_segmentation",
+                                         "LDA_preprocessing",
```
Comment (ngreenwald) [inline ark/utils/example_dataset_test.py:15]: Do each of these need a different example dataset? I think the LDA training and inference notebook is the only one that has specific preprocessing

**Review by ngreenwald — CHANGES_REQUESTED** [review]
We should be able to delete all of the files in /example_data now

File: templates/4_Post_Clustering.ipynb [hunk 2]
```diff
@@ -48,7 +48,7 @@
   },
```
Comment (ngreenwald) [inline templates/4_Post_Clustering.ipynb:48]: <p>We can clear this stuff</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/angelolab/ark-analysis/pull/832/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

File: templates/example_neighborhood_analysis_script.ipynb [hunk 3]
```diff
@@ -18,7 +18,7 @@
   },
```
Comment (ngreenwald) [inline templates/example_neighborhood_analysis_script.ipynb:18]: <p>We should clear all the output of uninformative cells (i.e. keep the plots, remove everything else)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/angelolab/ark-analysis/pull/832/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

Comment (ngreenwald) [inline templates/example_neighborhood_analysis_script.ipynb:18]: <p>Just gold/Au</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/angelolab/ark-analysis/pull/832/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='29'/>

File: ark/utils/example_dataset_test.py (hunk 1 above)
Comment (srivarra) [reply ark/utils/example_dataset_test.py:15]: The datasets themselves are the same, they just have different configurations. Each one only uses the minimal amount of data required.

File: templates/example_neighborhood_analysis_script.ipynb (hunk 3 above)
Comment (srivarra) [reply templates/example_neighborhood_analysis_script.ipynb:18]: <p>Au doesn't exist in the Example Dataset.</p>

Comment (ngreenwald) [reply templates/example_neighborhood_analysis_script.ipynb:18]: Nope, in that case we don't need to exclude anything

File: ark/utils/example_dataset_test.py (hunk 1 above)
Comment (ngreenwald) [reply ark/utils/example_dataset_test.py:15]: Got it, sounds good

**Pull request merged**